### PR TITLE
Nat support for config & show commands 

### PR DESCRIPTION
**Pull request by ArthiSivanantham**
This PR contains,
1) subtree transformer for NAT/NAPT Tables
2) Bug fixes for NAT Pool/Binding Configurations
3) Show all NAT Translation entries

### CODE REVIEW

File: src/translib/transformer/xfmr_nat.go [hunk 1]
```diff
@@ -167,103 +167,1033 @@ func findProtocolByValue(m map[uint8]string, value string) uint8 {
...
+            err = errors.New("URI data (NAT_TRANSLATIONS_BYTES) not found in db , " + attrUri)
+        }
+    case "/openconfig-nat:nat/instances/instance/nat-mapping-table/nat-mapping-entry/state/counters/dnat-translations-pkts":
+        if entry.Has("NAT_TRANSLATIONS_PKTS") {
+            natCntObj.NatTranslationsPkts = new(uint64)
+            *natCntObj.NatTranslationsPkts , _ = strconv.ParseUint(entry.Get("NAT_TRANSLATIONS_PKTS"), 10, 64)
+        } else {
+            err = errors.New("URI data (NAT_TRANSLATIONS_PKTS) not found in db , " + attrUri)
+        }
+        /*
```
Comment (rvasanthm) [inline src/translib/transformer/xfmr_nat.go]: Delete commented code

File: src/translib/transformer/xfmr_nat.go [hunk 2]
```diff
@@ -167,103 +167,1033 @@ func findProtocolByValue(m map[uint8]string, value string) uint8 {
...
+        }
+
+        dbkey := key
+        if _, ok := natMap[tblName]; !ok {
+            natMap[tblName] = make (map[string]db.Value)
+        }
+
+        dbData := make(map[string]string)
+        entry := db.Value{Field: dbData}
+        if data.Config != nil {
```
Comment (rvasanthm) [inline src/translib/transformer/xfmr_nat.go:312]: Indentation

File: src/translib/transformer/xfmr_nat.go [hunk 3]
```diff
@@ -167,103 +167,1033 @@ func findProtocolByValue(m map[uint8]string, value string) uint8 {
...
+        //dbkey := db.Key{Comp: []string{key.ExternalAddress, protocol_map[key.Protocol)], strconv.FormatInt(int64(key.ExternalPort), 10)}
+        dbkey := key.ExternalAddress + "|" + protocol_map[key.Protocol] + "|" + strconv.FormatInt(int64(key.ExternalPort), 10)
+        if _, ok := naptMap[tblName]; !ok {
+            naptMap[tblName] = make (map[string]db.Value)
+        }
+
+        dbData := make(map[string]string)
+        entry := db.Value{Field: dbData}
+
+        if data.Config != nil {
```
Comment (rvasanthm) [inline src/translib/transformer/xfmr_nat.go:709]: Indentaion

File: src/translib/transformer/xfmr_nat.go [hunk 4]
```diff
@@ -167,103 +167,1033 @@ func findProtocolByValue(m map[uint8]string, value string) uint8 {
...
+        }
+    case "/openconfig-nat:nat/instances/instance/napt-mapping-table/napt-mapping-entry/state/counters/nat-translations-pkts":
+        if entry.Has("NAT_TRANSLATIONS_PKTS") {
+            naptCntObj.NatTranslationsPkts = new(uint64)
+            *naptCntObj.NatTranslationsPkts , _ = strconv.ParseUint(entry.Get("NAT_TRANSLATIONS_PKTS"), 10, 64)
+        } else {
+            err = errors.New("URI data (NAT_TRANSLATIONS_PKTS) not found in db , " + attrUri)
+        }
+        /*
+    case "/openconfig-nat:nat/instances/instance/napt-mapping-table/napt-mapping-entry/state/counters/snat-translations-bytes":
```
Comment (rvasanthm) [inline src/translib/transformer/xfmr_nat.go]: Delete commented code

File: src/translib/transformer/xlate_from_db.go [hunk 5]
```diff
@@ -631,7 +631,9 @@ func yangListInstanceDataFill(dbs [db.MaxDB]*db.DB, ygRoot *ygot.GoStruct, uri s
 				log.Infof("Error returned by %v: %v", xYangSpecMap[xpath].xfmrFunc, err)
```
Comment (rvasanthm) [inline src/translib/transformer/xlate_from_db.go]: Why are you committing transformer changes as part of this pull request?

File: src/translib/transformer/xfmr_nat.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ArthiSivanantham) [reply src/translib/transformer/xfmr_nat.go]: Addressed

Comment (ArthiSivanantham) [reply src/translib/transformer/xfmr_nat.go]: Addressed

File: src/translib/transformer/xfmr_nat.go (hunk 3 above)
Comment (ArthiSivanantham) [reply src/translib/transformer/xfmr_nat.go:709]: Addressed

File: src/translib/transformer/xlate_from_db.go (hunk 5 above)
Comment (ArthiSivanantham) [reply src/translib/transformer/xlate_from_db.go]: As mentioned in the description, This PR uses fixes from "transformer-phase2" branch. We'll hold merge of this PR  until "transformer-phase2" is merged into "dell_sonic" branch.

**Review by bhavini-gada — APPROVED** [review]

**Pull request merged**